### PR TITLE
Dynamic lookup of GDPR-related modules for retrieval

### PR DESCRIPTION
**Pull request by fenek**
This PR replaces hardcoded GDPR modules list with a dynamic lookup. It also changes the retrieval implementations to query all module backends.

- [x] Implementation
- [x] Tests?

### CODE REVIEW

**Comment by mongoose-im** [comment]
[6441.1](https://travis-ci.org/esl/MongooseIM/jobs/529836334) / Erlang 19.3 / small_tests / a1fae11266b4c7a6bbf51abd3cef2389dca51661
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2297/6441/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2297/6441/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-05-08_18.25.47/lib.mongooseim.logs/run.2019-05-08_18.25.48/suite.log.html)

**Comment by mongoose-im** [comment]
[6444.1](https://travis-ci.org/esl/MongooseIM/jobs/530169614) / Erlang 19.3 / small_tests / d55510e91b4c9b43058996bc81c18bbaa50c67d0
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2297/6444/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2297/6444/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-05-09_08.52.30/lib.mongooseim.logs/run.2019-05-09_08.52.31/suite.log.html)

File: src/mod_vcard.erl [hunk 1]
```diff
@@ -134,12 +134,17 @@ get_personal_data(Username, Server) ->
...
-         _ -> []
-        end,
+    Entries = lists:flatmap(fun(B) ->
+                                    try B:get_vcard(LUser, LServer) of
+                                        {ok, Record} ->
+                                            SerializedRecords = exml:to_binary(Record),
+                                            [{Jid, SerializedRecords}];
+                                        _ -> []
+                                    catch
+                                        _:_ ->
```
Comment (erszcz) [inline src/mod_vcard.erl]: Silent error - is there really a reason not to log it? This will be a nightmare to debug.

Comment (fenek) [reply src/mod_vcard.erl]: I thought about the same thing after merging and decided to fix it after all other PRs are merged.

File: src/pubsub/mod_pubsub.erl [hunk 2]
```diff
@@ -262,14 +262,24 @@ process_packet(Acc, From, To, El, #state{server_host = ServerHost, access = Acce
...
      [{pubsub_payloads, ["node_name", "item_id", "payload"], Payloads},
       {pubsub_nodes, ["node_name", "type"], Nodes},
       {pubsub_subscriptions, ["node_name"], Subscriptions}].
 
+get_personal_data_group(LUser, LServer, Backends, FunctionName) ->
+    lists:flatmap(fun(B) -> try B:FunctionName(LUser, LServer) of
+                                Result when is_list(Result) -> Result;
+                                _ -> []
+                            catch
+                                _:_ -> []
```
Comment (erszcz) [inline src/pubsub/mod_pubsub.erl]: FYI, a similar silent catch-all

**Pull request merged**